src/lib_ccx/hardsubx_russian.c

```diff
@@ -0,0 +1,995 @@
+	for(i=(92*height)/100;i<height;i++)
+	{
+		for(j=0;j<width;j++)
+		{
+			int p=j*3+i*frame->linesize[0];
+			int r=frame->data[0][p];
+			int g=frame->data[0][p+1];
+			int b=frame->data[0][p+2];
+			pixSetRGBPixel(im,j,i,r,g,b);
+			float L,A,B;
+			rgb2lab((float)r,(float)g,(float)b,&L,&A,&B);
+			if(L > 85)
+				pixSetRGBPixel(lum_im,j,i,255,255,255);
+			else
+				pixSetRGBPixel(lum_im,j,i,0,0,0);
+		}
+	}
+
+
+	//Handle the edge image
+	edge_im = pixCreate(width,height,8);
+	edge_im = pixConvertRGBToGray(im,0.0,0.0,0.0);
+	edge_im = pixSobelEdgeFilter(edge_im, L_VERTICAL_EDGES);
+	edge_im = pixDilateGray(edge_im, 21, 11);
+	edge_im = pixThresholdToBinary(edge_im,50);
+
+	for(i=92*(height/100);i<height;i++)
+	{
+		for(j=0;j<width;j++)
+		{
+			unsigned int p1,p2,p3;
+			pixGetPixel(edge_im,j,i,&p1);
+			// pixGetPixel(pixd,j,i,&p2);
+			pixGetPixel(lum_im,j,i,&p3);
+			if(p1==0&&p3>0)
+				pixSetRGBPixel(feat_im,j,i,255,255,255);
+			else
+				pixSetRGBPixel(feat_im,j,i,0,0,0);
+		}
+	}
+
+
+
+
+
+       int num_of_occr = 0;
+       int stop = 0;
+       char write_path[500];
+       for(i=(92*height)/100;i<height && !stop;i++){
+            for(j=0; j<width  && !stop; j++){
+ 
+                 int p=j*3+i*frame->linesize[0];
+			int r=frame->data[0][p];
+			int g=frame->data[0][p+1];
+			int b=frame->data[0][p+2];
+
+
+        if (detect_stopper_color(r,g,b,ctx) == 1 && (index == ctx->start_frame ||index % ctx->frame_skip == 0) && t_flag == 0){
+         //printf("%s\n", "Stopper Detected");  
+         xflag = 1;
+         t_flag = 1;
+         continue_flag = 0;
+         //Hardcoded Bouding Box Locations
+         BOX* box = boxCreate(100, 525, 615, 25);
+         PIX* pixd= pixClipRectangle(feat_im, box, NULL);
+         last_index = index;
+         pix_temp = pixCopy(NULL, pixd);
+         subtitle_text = russian_ocr(pixd, index);
+         pixGetDimensions(pixd, &w, NULL, &d);
+         pre_index = index;
+
+
+
+
+         //sprintf(write_path,"im%04d.jpg",index);
+         //pixWrite(write_path,pixd,IFF_JFIF_JPEG);
+         boxDestroy(&box);
+	 pixDestroy(&pixd);
+         stop = 1;
+         }
+
+         else if (detect_stopper_color(r,g,b,ctx) == 1 && (index > (pre_index + 15)) && t_flag == 1){
+         //printf("%s\n", "Stopper Detected");  
+
+         if (continue_flag == 0){
+         t_flag = 0;
+         printf("%s", "Here");
+         //Hardcoded Bouding Box Locations
+         BOX* box = boxCreate(100, 525, 615, 25);
+         PIX* pixd= pixClipRectangle(feat_im, box, NULL);
+         last_index = index;
+         //subtitle_text = russian_ocr(pixd, index);
+         printf("%f", check_frames(pixd, pix_temp));
+
+         if (check_frames(pixd, pix_temp) <= 30){
+         t_flag = 0;
+         sprintf(write_path,"im%04d.jpg",pre_index);
+         if (count_frame % 2 == 0) {
+         pixWrite(write_path,pix_temp,IFF_JFIF_JPEG);
+         subtitle_text = russian_ocr(pixd, index);
```

Comment:
Please write to file only in debug mode (within #ifdef DEBUG). Also, do not use printf anywhere for non useful information ("Here"). To print useful information, use mprint() as elsewhere in the code.

Reply:
Okay, Noted :)